elastic/kibana
# [Log Explorer] Show only dataset name in dataset selector

closes https://github.com/elastic/kibana/issues/166247

## 📝  Summary

This PR changes the display of datasets in the dataset selector so that we only show the dataset name and not the whole index pattern.

ex: 

- `logs-foo-*` ➡️ `foo`
- `logs-bar-*` ➡️ `bar`

## ✅  Testing
1. Navigate to the observability log explorer
2. Click the dataset selector and choose Uncategorized
3. Observe how the datasets are represented

## 🎥 Demo

https://github.com/elastic/kibana/assets/11225826/8a6a1212-46d5-4ceb-bf38-cf13d1b3e555
